python/jupytergis_core/jupytergis_core/handler.py

```diff
@@ -0,0 +1,35 @@
+import json
+from jupyter_server.base.handlers import APIHandler
+from jupyter_server.utils import url_path_join
+import tornado
+import requests
+
+class ProxyHandler(APIHandler):
```

Comment:
Defined this proxy handler and a get method However an another failure persists at the moment i.e. (404 Not Found)
```
GET http://localhost:8888/jupytergis_core/proxy?url=https://docs.geoserver.org/stable/en/user/_downloads/30e405b790e068c43354367cb08e71bc/nyc_roads.zip 404 (Not Found)
```

Might be a possibility that I may not have implemented it 100% correctly as suggested but I'm still looking into it :)

Reply:
I was missing this only, It works perfectly now. CORS Error resolved 🎉. pheww
Thanks a lot @martinRenou

Reply:
Question: Would this proxy handler not work on jupyterlite deployment? And would the users also need to run `jupyter server extension enable jupytergis_core` locally once?

Reply:
> Would this proxy handler not work on jupyterlite deployment?

Unfortunately no, this is because we don't have a tornado server running in the case of JupyterLite (JupyterLite is just a bunch of assets HTML/CSS served by a static server).

> And would the users also need to run jupyter server extension enable jupytergis_core locally once?

They won't need to, just doing `pip/conda install jupytergis` will properly install [this file](https://github.com/QuantStack/jupytergis/pull/104/files#diff-c75ad23365ae06a833d22b84a92e25638a232688d8a85e0a83aab2b82431189e) thanks to [this line](https://github.com/QuantStack/jupytergis/pull/104/files#diff-12154add17eaf6a335d76b78722e3e58f080de5f5bec0b5aac9b4db9517f563dR46) (you set it up properly). At start up, the jupyter server will automatically look into this directory read `jupytergis_core.json`, see that `jupytergis_core` contains a server extension, and load it.

If you had to manually run this command, it's probable that it's because the `jupytergis_core.json` file was not installed in your case.

Reply:
> Unfortunately no, this is because we don't have a tornado server running in the case of JupyterLite (JupyterLite is just a bunch of assets HTML/CSS served by a static server).

I see, I was thinking about it (which might sound absurd) since JupyterLite's Python kernel is backed by [Pyodide](https://pyodide.org/). Could it be a possibility that if we manage to have the recipe of `tornado` package in pyodide(hypothetically), then we may be able to do this?

> They won't need to, just doing pip/conda install jupytergis will properly install [this file](https://github.com/QuantStack/jupytergis/pull/104/files#diff-c75ad23365ae06a833d22b84a92e25638a232688d8a85e0a83aab2b82431189e) thanks to [this line](https://github.com/QuantStack/jupytergis/pull/104/files#diff-12154add17eaf6a335d76b78722e3e58f080de5f5bec0b5aac9b4db9517f563dR46) (you set it up properly). At start up, the jupyter server will automatically look into this directory read jupytergis_core.json, see that jupytergis_core contains a server extension, and load it.

> 
> If you had to manually run this command, it's probable that it's because the jupytergis_core.json file was not installed in your case.

Got it. Thanks for such a comprehensive explanation 🚀

Reply:
> I see, I was thinking about it (which might sound absurd) since JupyterLite's Python kernel is backed by [Pyodide](https://pyodide.org/). Could it be a possibility that if we manage to have the recipe of tornado package in pyodide(hypothetically), then we may be able to do this?

We thought about this kind of approach for other tasks (like have a in-browser Python kernel render voila dashboards on the fly). I feel like we should avoid this approach as much as we can, having a in-browser Python kernel is a bit CPU and memory intensive.

Reply:
> We thought about this kind of approach for other tasks (like have a in-browser Python kernel render voila dashboards on the fly). I feel like we should avoid this approach as much as we can, having a in-browser Python kernel is a bit CPU and memory intensive.

That makes sense. Then I believe a custom proxy server can be used in this case too for JupyterLite as discussed in #93